baselines/niid_bench/README.md

```diff
@@ -0,0 +1,131 @@
+
+| Dataset | #classes | #partitions | partitioning method | partition settings |
+| :------ | :---: | :---: | :---: | :---: |
+| CIFAR10 | 10 | 10 | IID | NA |
+| CIFAR10 | 10 | 10 | dirichlet | distribution parameter 0.5 |
+| CIFAR10 | 10 | 10 | sort and partition | 1 label per client |
+| CIFAR10 | 10 | 10 | sort and partition | 2 labels per client |
+| CIFAR10 | 10 | 10 | sort and partition | 3 labels per client |
+| MNIST | 10 | 10 | IID | NA |
+| MNIST | 10 | 10 | dirichlet | distribution parameter 0.5 |
+| MNIST | 10 | 10 | sort and partition | 1 label per client |
+| MNIST | 10 | 10 | sort and partition | 2 labels per client |
+| MNIST | 10 | 10 | sort and partition | 3 labels per client |
+| FMNIST | 10 | 10 | IID | NA |
+| FMNIST | 10 | 10 | dirichlet | distribution parameter 0.5 |
+| FMNIST | 10 | 10 | sort and partition | 1 label per client |
+| FMNIST | 10 | 10 | sort and partition | 2 labels per client |
+| FMNIST | 10 | 10 | sort and partition | 3 labels per client |
+
+
+****Training Hyperparameters:**** There are four FL algorithms and they have many common hyperparameters and few different ones. The following table shows the common hyperparameters and their default values.
+
+| Description | Default Value |
+| ----------- | ----- |
+| total clients | 10 |
+| clients per round | 10 |
+| number of rounds | 50 |
+| number of local epochs | 10 |
+| client resources | {'num_cpus': 4.0, 'num_gpus': 0.0 }|
+| dataset name | cifar10 
+| data partition | Dirichlet (0.5) |
+| batch size | 64 |
+| momentum for SGD | 0.9 |
+
+For FedProx algorithm the proximal parameter is tuned from values {0.001, 0.01, 0.1, 1.0} in the experiments. The default value is 0.01. 
+
+
+## Environment Setup
+
+```bash
+# Setup the base poetry enviroment from the niid_bench directory
+poetry install
+
+# Start the shell
+poetry shell
+
+#  Install tqdm, torch, and torchvision using pip
+pip install torch==1.12.1+cu113 torchvision==0.13.1+cu113 --extra-index-url https://download.pytorch.org/whl/cu113
+pip install tqdm
+```
+
+
+## Running the Experiments
+You can run four algorithms fedavg, scaffold, fedprox, and fednova. To run any of them, use any of the corresponding main files. For instance, the following command will run with the default config provided in the corresponding configuration files.
+
+```bash
+python -m niid_bench.main_fedprox
```

Comment:
I haven't looked in absolute detail but at a first glance, it seems all `main_` files are very very similar. Would it be sufficient to have a single `main.py` and then pass a different config depending on what strategy we want to run?
You can specify which config to use with Hydra's `--config-name` argument. Like this:

```
# this runs the content of `main_fedavg.py` but changes the default config (`conf/fedavg_base.yaml`) to `conf/fednova_base`
python -m niid_bench.main_fedavg --config-name fednova_base
```

Reply:
You know your code better, but if this is possible, it will be ideal to prevent unnecessary code duplication.